### PR TITLE
deflakes test-blacklisted-instance-not-reserved

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- deflakes test-blacklisted-instance-not-reserved

## Why are we making these changes?

It fixes how we compute the `max-backoff-exponent`.
We would like all our tests to run and catch regressions.

### CODE REVIEW

**Comment by shamsimam** [comment]
@pschorf any feedback on this?

File: waiter/src/waiter/state.clj [hunk 1]
```diff
@@ -539,7 +539,11 @@
    initial-state]
   (when (some nil? (vals initial-state))
     (throw (ex-info "Initial state contains nil values!" initial-state)))
-  (let [max-backoff-exponent (int (max 1 (inc (/ (Math/log max-blacklist-time-ms) (Math/log blacklist-backoff-base-time-ms)))))
+  (let [max-backoff-exponent (->> (/ (Math/log max-blacklist-time-ms) (Math/log blacklist-backoff-base-time-ms))
+                                  Math/exp
```
Comment (pschorf) [inline waiter/src/waiter/state.clj]: It doesn't look like this was in the original code, was it a bug that it was missing?

File: waiter/integration/waiter/killed_instance_test.clj [hunk 2]
```diff
@@ -65,50 +43,73 @@
...
+                        (wait-for :interval 5 :timeout 10))))
+                (keys router-id->router-url))]
+          (doseq [[router-id _] router-id->router-url]
+            (is (router-id->blacklist-time router-id)
+                (str "instance not blacklisted " {:instance-id instance-id :router-id router-id})))
+
+          (let [new-request-instance-ids-atom (atom #{})]
+            (parallelize-requests
+              10 1
+              #(let [{:keys [instance-id router-id]} (make-request-fn)]
```
Comment (pschorf) [inline waiter/integration/waiter/killed_instance_test.clj]: I'd rather use `fn` here, I kept looking for the `%` inside the code.

File: waiter/src/waiter/state.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/state.clj]: Yes, this was a bug that got fixed.

File: waiter/integration/waiter/killed_instance_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/integration/waiter/killed_instance_test.clj]: Okay, will do.

File: waiter/src/waiter/state.clj (hunk 1 above)
Comment (pschorf) [reply waiter/src/waiter/state.clj]: 👍

File: waiter/integration/waiter/killed_instance_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/integration/waiter/killed_instance_test.clj]: Done.

**Comment by shamsimam** [comment]
Ready for review.

**Pull request merged**